### PR TITLE
php: Multiple package and extension updates

### PR DESCRIPTION
**Pull request by etu**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by aanderse — APPROVED** [review]

File: pkgs/development/php-packages/imagick/default.nix [hunk 1]
```diff
@@ -3,21 +3,8 @@
 buildPecl {
   pname = "imagick";
 
-  version = "3.4.4";
-  sha256 = "0xvhaqny1v796ywx83w7jyjyd0nrxkxf34w9zi8qc8aw8qbammcd";
-
-  patches = [
-    # Fix compatibility with PHP 8.
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/imagick/default.nix:11]: Please remove the unused fetchpatch from inputs.

**Comment by risicle** [comment]
> 3.4.4 -> 3.5.0
> 2.18.4 -> 3.0.0
> 4.6.1 -> 4.7.3

I've really got to question porting some of these bumps back to the "stable" release. What is the purpose of a stable release if it's going to receive random surprises? Why not just track `master` if people want to keep up with all these versions?

**Comment by aanderse** [comment]
I don't get it... this targets `master`.

**Comment by risicle** [comment]
The backport-release-21.05 label will automatically create a backport PR for 21.05 once this is merged.

**Comment by aanderse** [comment]
Oops! Sorry I didn't see the label.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128087` at 646a411c run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built successfully:</summary>
  <ul>
    <li>php74Extensions.event</li>
    <li>php74Extensions.igbinary</li>
    <li>php74Extensions.imagick</li>
    <li>php74Extensions.protobuf</li>
    <li>php74Packages.php-cs-fixer</li>
    <li>php74Packages.phpstan</li>
    <li>php74Packages.psalm</li>
    <li>php80Extensions.event</li>
    <li>php80Extensions.igbinary</li>
    <li>php80Extensions.imagick</li>
    <li>php80Extensions.protobuf</li>
    <li>php80Packages.php-cs-fixer</li>
    <li>php80Packages.phpstan</li>
    <li>php80Packages.psalm</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/php-packages/phpstan/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/php-packages/php-cs-fixer/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/php-packages/psalm/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by etu** [comment]
I've looked through the release notes of php-cs-fixer which was the only
major version bump.

I don't know why they consider it a major version bump after reading that
because all I could find was big fixes and things that could fit in a minor
versions bump.

https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/tag/v3.0.0

Also, backporting and what to backport is a separate discussion that
belongs to that PR thread if you ask me.

File: pkgs/development/php-packages/imagick/default.nix (hunk 1 above)
Comment (etu) [reply pkgs/development/php-packages/imagick/default.nix:11]: Removed

**Comment by jtojnar** [comment]
> I don't know why they consider it a major version bump after reading that because all I could find was big fixes and things that could fit in a minor versions bump.

There are some BC breaks in option names and values. Also minimal PHP version has changed.

**Comment by etu** [comment]
Then the person merging this should feel free to remove the backport label
before merging :)

**Pull request merged**